src/viam/ur/module/ur_arm.cpp

```diff
@@ -185,9 +203,35 @@ std::vector<std::shared_ptr<ModelRegistration>> URArm::create_model_registration
 }
 
 URArm::URArm(Model model, const Dependencies& deps, const ResourceConfig& cfg) : Arm(cfg.name()), model_(std::move(model)) {
-    VIAM_SDK_LOG(info) << "URArm constructor start (model: " << model_.to_string() << ")";
+    VIAM_SDK_LOG(info) << "URArm constructor called (model: " << model_.to_string() << ")";
+    startup_(deps, cfg);
+}
+
+void URArm::startup_(const Dependencies&, const ResourceConfig& cfg) {
+    if (current_state_) {
+        throw std::logic_error("URArm::startup_ was called for an already running instance");
+    }
+
+    VIAM_SDK_LOG(info) << "URArm starting up";
     current_state_ = std::make_unique<state_>();
-    this->reconfigure(deps, cfg);
+
+    // If we fail to make it through the startup sequence, execute the shutdown code. The
+    // shutdown code must be prepared to be called from any intermediate state that this
+    // function may have constructed due to partial execution.
+    auto failure_handler = make_scope_guard([&] {
+        VIAM_SDK_LOG(warn) << "URArm startup failed - shutting down";
+        shutdown_();
+    });
+
+    // extract relevant attributes from config
+    current_state_->host = find_config_attribute<std::string>(cfg, "host");
+    current_state_->speed.store(find_config_attribute<double>(cfg, "speed_degs_per_sec") * (M_PI / 180.0));
+    current_state_->acceleration.store(find_config_attribute<double>(cfg, "acceleration_degs_per_sec2") * (M_PI / 180.0));
+    try {
+        const std::lock_guard<std::mutex> guard{current_state_->output_csv_dir_path_mu};
+        current_state_->output_csv_dir_path = find_config_attribute<std::string>(cfg, "csv_output_path");
+    } catch (...) {  // NOLINT: TODO: What should actually happen if the attribute is missing?
```

Comment:
below this we set `output_csv_dir_path` to whatever is in `VIAM_MODULE_DATA` if output_csv_dir_path is empty. maybe we can just move that logic into the catch?

Reply:
This code didn't change in this review. It just moved. I'm trying to make this review as close to simple code motion as I can.